rotkehlchen/tests/unit/decoders/test_velodrome.py

```diff
@@ -0,0 +1,832 @@
+            counterparty=CPT_VELODROME,
+            address=ROUTER_V2,
+            notes=F'Receive 0.0770899907450713 ETH as the result of a swap in {CPT_VELODROME}',
+        ),
+    ]
+    assert events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('optimism_accounts', [['0xeEf53a1f4eb3846f33C3E549D6FDF130fa4f8b27']])
+def test_swap_token_to_eth_v1(optimism_accounts, optimism_transaction_decoder):
+    """Check that swapping token to eth in velodrome is properly decoded."""
+    evmhash = deserialize_evm_tx_hash('0x10d52ec5cf06d1b5b48a346334c8f0e6ef83bd281c371d762d3c163e37f629d8')  # noqa: E501
+    user_address = optimism_accounts[0]
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=optimism_transaction_decoder.evm_inquirer,
+        database=optimism_transaction_decoder.database,
+        tx_hash=evmhash,
+    )
+    timestamp = TimestampMS(1695155023000)
+    expected_events = [
+        EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.OPTIMISM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(FVal('0.000076204005061914')),
+            location_label=user_address,
+            counterparty=CPT_GAS,
+            notes='Burned 0.000076204005061914 ETH for gas',
+        ), EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=1,
+            timestamp=timestamp,
+            location=Location.OPTIMISM,
+            event_type=HistoryEventType.TRADE,
+            event_subtype=HistoryEventSubType.SPEND,
+            asset=Asset(VELO_V1_TOKEN),
+            balance=Balance(FVal('5165.602591359381942771')),
+            location_label=user_address,
+            counterparty=CPT_VELODROME,
+            address=string_to_evm_address('0xe8537b6FF1039CB9eD0B71713f697DDbaDBb717d'),  # velo_usdc_pool_address  # noqa: E501
+            notes=f'Swap 5165.602591359381942771 VELO in {CPT_VELODROME}',
+        ), EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=2,
+            timestamp=timestamp,
+            location=Location.OPTIMISM,
+            event_type=HistoryEventType.TRADE,
+            event_subtype=HistoryEventSubType.RECEIVE,
+            asset=A_ETH,
+            balance=Balance(FVal('0.181508840163183003')),
+            location_label=user_address,
+            counterparty=CPT_VELODROME,
+            address=ROUTER_V1,
+            notes=F'Receive 0.181508840163183003 ETH as the result of a swap in {CPT_VELODROME}',
+        ),
+    ]
+    assert events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('optimism_accounts', [['0x78C13393Aee675DD7ED07ce992210750D1F5dB88']])
+def test_swap_tokens(optimism_accounts, optimism_transaction_decoder):
+    """Check that swapping tokens in velodrome is properly decoded."""
```

Comment:
same as above here and in other places. Specify version. Let's be future proof (maybe a v3 comes along etc.)

Check rest of the test functions and do that there too. I won't leave more comments in the PR for this.